lib/pure/times.nim

```diff
@@ -744,6 +744,336 @@ proc format*(info: TimeInfo, f: string): string =
+            info.month =  mJan
+            j += 7
+        elif value.match(peg"^i'february'.*"):
+            info.month =  mFeb
+            j += 8
+        elif value.match(peg"^i'march'.*"):
+            info.month =  mMar
+            j += 5
+        elif value.match(peg"^i'april'.*"):
+            info.month =  mApr
+            j += 5
+        elif value.match(peg"^i'may'.*"):
+            info.month =  mMay
+            j += 3
+        elif value.match(peg"^i'june'.*"):
+            info.month =  mJun
+            j += 4
+        elif value.match(peg"^i'july'.*"):
+            info.month =  mJul
+            j += 4
+        elif value.match(peg"^i'august'.*"):
+            info.month =  mAug
+            j += 6
+        elif value.match(peg"^i'september'.*"):
+            info.month =  mSep
+            j += 9
+        elif value.match(peg"^i'october'.*"):
+            info.month =  mOct
+            j += 7
+        elif value.match(peg"^i'november'.*"):
+            info.month =  mNov
+            j += 8
+        elif value.match(peg"^i'december'.*"):
+            info.month =  mDec
+            j += 8
+        else:
+            raise newException(ValueError, "invalid month") 
+    of "s":
+        if value[j+1] in Digits:
+            # two digit format
+            info.second = value[j..j+1].parseInt()
+            j += 2
+        else:
+            info.second = parseInt($value[j])
+            j += 1
+    of "ss":
+        # two digit format
+        info.second = value[j..j+1].parseInt()
+        j += 2
+    of "t":
+        if value[j] == 'P' and info.hour > 0 and info.hour < 12:
+            info.hour += 12
+        j += 1
+    of "tt":
+        if value[j..j+1] == "PM" and info.hour > 0 and info.hour < 12:
+            info.hour += 12
+        j += 2
+    of "yy":
+        # Assumes current century
+        var year = value[j..j+1].parseInt()
+        var thisCen = getLocalTime(getTime()).year div 100
+        info.year = thisCen*100 + year
+        j += 2
+    of "yyyy":
+        info.year = value[j..j+3].parseInt()
+        j += 4
+    of "z":
+        if value[j] == '+':
+            info.timezone = parseInt($value[j+1])
+        elif value[j] == '-':
+            info.timezone = 0-parseInt($value[j+1])
+        else:
+            raise newException(ValueError, "Sign for timezone " & value[j])
+        j += 2
+    of "zz":
+        if value[j] == '+':
+            info.timezone = value[j+1..j+2].parseInt()
+        elif value[j] == '-':
+            info.timezone = 0-value[j+1..j+2].parseInt()
+        else:
+            raise newException(ValueError, "Sign for timezone " & value[j])
+        j += 3
+    of "zzz":
+        if value[j] == '+':
+            info.timezone = value[j+1..j+2].parseInt()
+        elif value[j] == '-':
+            info.timezone = 0-value[j+1..j+2].parseInt()
+        else:
+            raise newException(ValueError, "Sign for timezone " & value[j])
+        j += 6
```

Comment:
btw, not that I really case, but the rest of the file uses 2 spaces for indentation.